include/moqt/core/transport.h

```diff
@@ -0,0 +1,226 @@
+
+            uint64_t current_subscribe_id{ 0 }; ///< Connection specific ID for subscribe messages
+
+            /// Track namespace/name by received subscribe IDs
+            /// Used to map published tracks to subscribes in client mode
+            std::map<uint64_t, std::pair<uint64_t, uint64_t>> recv_sub_id;
+
+            /// Tracks by subscribe ID
+            std::map<uint64_t, std::shared_ptr<BaseTrackHandler>> tracks_by_sub_id;
+
+            /// Publish tracks by namespace and name. map[track namespace][track name] = track delegate
+            std::map<uint64_t, std::map<uint64_t, std::shared_ptr<BaseTrackHandler>>> pub_tracks_by_name;
+        };
+
+        // -------------------------------------------------------------------------------------------------
+        // Private methods
+        // -------------------------------------------------------------------------------------------------
+        void Init();
+
+        void SendCtrlMsg(const ConnectionContext& conn_ctx, std::vector<uint8_t>&& data);
+        void SendClientSetup();
+        void SendServerSetup(ConnectionContext& conn_ctx);
+        void SendAnnounce(ConnectionContext& conn_ctx, Span<uint8_t const> track_namespace);
+        void SendAnnounceOk(ConnectionContext& conn_ctx, Span<uint8_t const> track_namespace);
+        void SendUnannounce(ConnectionContext& conn_ctx, Span<uint8_t const> track_namespace);
+        void SendSubscribe(ConnectionContext& conn_ctx, uint64_t subscribe_id, FullTrackName& tfn, TrackHash th);
+        void SendSubscribeOk(ConnectionContext& conn_ctx, uint64_t subscribe_id, uint64_t expires, bool content_exists);
+        void SendUnsubscribe(ConnectionContext& conn_ctx, uint64_t subscribe_id);
+        void SendSubscribeDone(ConnectionContext& conn_ctx, uint64_t subscribe_id, const std::string& reason);
+        void SendSubscribeError(ConnectionContext& conn_ctx,
+                                uint64_t subscribe_id,
+                                uint64_t track_alias,
+                                messages::MoqtSubscribeError error,
+                                const std::string& reason);
+        void CloseConnection(TransportConnId conn_id,
+                             messages::MoqtTerminationReason reason,
+                             const std::string& reason_str);
+        bool ProcessRecvCtrlMessage(ConnectionContext& conn_ctx, std::shared_ptr<StreamBuffer<uint8_t>>& stream_buffer);
+        bool ProcessRecvStreamDataMessage(ConnectionContext& conn_ctx,
+                                          std::shared_ptr<StreamBuffer<uint8_t>>& stream_buffer);
+
+        void RemoveSubscribeTrack(ConnectionContext& conn_ctx,
+                                  SubscribeTrackHandler& delegate,
+                                  bool remove_delegate = true);
+
+        std::optional<std::weak_ptr<PublishTrackHandler>> GetPubTrackHandler(ConnectionContext& conn_ctx,
```

Comment:
given it is a weak pointer, i don;t see it needs to be inside an optional.

Reply:
Again, this is not part of the public API as these are private members used by the implementation.